src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/LeftCurlyCheck.java

```diff
@@ -495,7 +496,10 @@ else if (!TokenUtil.areOnSameLine(startToken, brace)) {
      * @param braceLine line content
      */
     private void validateEol(DetailAST brace, String braceLine) {
-        if (CommonUtil.hasWhitespaceBefore(brace.getColumnNo(), braceLine)) {
+        final String line = Pattern.compile("/\\*.*?\\*/")
```

Comment:
Let's name this a bit more descriptively, to make our intention obvious: `braceLineNoComments` or something.

Reply:
Also, let's find a more robust regexp for comments, maybe something like https://stackoverflow.com/questions/1657066/java-regular-expression-finding-comments-in-code

Reply:
Why we think that start of comment will be on same line .
Make test as
```
public Node(int value, Node next) /* 
      comment */ {
```
To be more confident that regexp solution is hack to cover specific case but not resolve a problem

Reply:
@romani I have think if we have to implement this https://github.com/checkstyle/checkstyle/pull/12382/files#r1026608868 we have to change the approach because currently we have the specific line only which  contains Left curly https://github.com/checkstyle/checkstyle/blob/a36dbb1a451e22b6349c84b98db58cfe15900155/src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/LeftCurlyCheck.java#L497

Reply:
Good point, @Kevin222004 let's extend regex to be something like `(?s)/\*.*?\*/|^(?!.*/\*).*\*/.*`, See https://regex101.com/r/5yflv3/1

First part is `?s)/\*.*?\*/`, this will match whole comments, and second part is `^(?!.*/\*).*\*/.*`, where we match comment started on some previous line.

Reply:
@nrmancuso This is cool but this case will fail here https://github.com/checkstyle/checkstyle/blob/7e0f1533d39c2811bd7804c5163d4d3b7fc2cd6b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocStyleCheck.java#L547-L550 
look here 
https://regex101.com/r/BfrEpY/1
🙃

Reply:
>I guess the only thing is know to approach with ast

Yes, we cannot have this regression. Let's not waste any more time with regexp. @romani do you want to create a new issue for rewrite of LeftCurlyCheck?

Reply:
@Kevin222004 after discussion with @romani , we have decided it is better to rewrite check, in the scope of a new issue. Please do:

1. Create new issue for check rewrite to use only AST, describe problems with regex approach (in general, not just what we dealt with in this PR)
2. Link to this PR, your test cases are good and we do not want to lose them.
3. Close this PR.

Thanks a lot for trying to fix this issue, sadly I do not think we can formulate a reliable regexp to fix this and cause no regression in the check.

Reply:
We can preserve all tests with comments.
And just merge this PR with only update for tests.
All tests will have `// ok, until #11410`
In PR commit should be `supplemental: tests improvement for #11411`

It is ok to keep issue the same, as for users nothing is going change for users.

Reply:
I have added few more test case which i have found from regression report

Reply:
and pgjdbc commit is also changed